photon-api/src/main/scala/com/linkedin/photon/ml/projector/ProjectionMatrixBroadcast.scala

```diff
@@ -72,6 +74,24 @@ protected[ml] class ProjectionMatrixBroadcast(projectionMatrixBroadcast: Broadca
     }
   }
 
+  /**
+   *
+   * @param randomEffectDataSet The input sharded data set in the original space
+   * @param normalizationContext The original broadcast form of NormalizationContext
+   * @return The broadcast form of NormalizationContext in projected space
+   */
+  def projectNormalizationContext(randomEffectDataSet: RandomEffectDataSet, normalizationContext: BroadcastWrapper[NormalizationContext])
+    : BroadcastWrapper[NormalizationContext] = {
+    val sc = randomEffectDataSet.sparkContext
+    val originalNormalizationContext = normalizationContext.value
+    val factors = originalNormalizationContext.factors.map(factors => projectionMatrixBroadcast.value.projectFeatures(factors))
+    val shifts = originalNormalizationContext.shifts.map(shifts => projectionMatrixBroadcast.value.projectFeatures(shifts))
+    val interceptId = originalNormalizationContext.interceptId.map(_ =>
+      projectionMatrixBroadcast.value.projectedSpaceDimension - 1)
```

Comment:
It seems odd that we'd need to know the intercept's position in the vector in this function. I think we should add a function / property to ProjectionMatrix that serves the same purpose as `projector.originalToProjectedSpaceMap(interceptId)` above.